Docs for VS Code extension (#5053)
sorbet/sorbet: website/blog/2022-01-06-open-sourcing-sorbet-vscode.md

---
id: open-sourcing-sorbet-vscode
title: Open-sourcing the Sorbet VS Code Extension
---

We’re excited to announce that Stripe’s VS Code extension for Sorbet is now
[open source](https://github.com/sorbet/sorbet/tree/master/vscode_extension).
We’ve designed Sorbet to be used in editors from Day 1—For the past two years,
Sorbet has exposed a flag (`--lsp`) that starts Sorbet in
[Language Server Protocol](https://microsoft.github.io/language-server-protocol/)
(LSP) mode. In this mode, Sorbet can respond to many LSP requests, like Go To
Definition, Find All References, Autocomplete, and more.

With this release, we’re making it even easier to take advantage of these LSP
features when working with Sorbet in VS Code. We hope that this extension takes
some of the guess work out of how to configure and use Sorbet’s LSP mode.

<!--truncate-->

Today’s release includes:

- The
  [pre-built extension](https://marketplace.visualstudio.com/items?itemName=sorbet.sorbet-vscode-extension)
  on the Visual Studio Marketplace

> - The [source code](https://github.com/sorbet/sorbet/issues/new/choose) for the

Reviewer: I believe this should rather link to https://github.com/sorbet/sorbet or https://github.com/sorbet/sorbet/tree/master/vscode_extension - I got pretty confused by being greeted with the "New Issue" view when trying to take a look at the code :)

---
id: open-sourcing-sorbet-vscode
title: Open-sourcing the Sorbet VS Code Extension
---

We’re excited to announce that Stripe’s VS Code extension for Sorbet is now

> [open source](http://#TODO). We’ve designed Sorbet to be used in editors from

Reviewer 2: ```suggestion
[open source](https://github.com/sorbet/sorbet/tree/master/vscode_extension). We’ve designed Sorbet to be used in editors from
```

> Day 1—Sorbet exposes a flag (`--lsp`) that starts Sorbet in

Reviewer 2: ```suggestion
Day 1—For the past two years, Sorbet has exposed a flag (`--lsp`) that starts Sorbet in
```

[Language Server Protocol](https://microsoft.github.io/language-server-protocol/)
(LSP) mode. In this mode, Sorbet can respond to many LSP requests, like Go To
Definition, Find All References, Autocomplete, and more.

With this release, we’re making it even easier to take advantage of these LSP
features when working with Sorbet in VS Code. We hope that this extension takes
some of the guess work out of how to configure and use Sorbet’s LSP mode.

<!--truncate-->

Today’s release includes:

> - The [pre-built extension](http://#TODO) on the Visual Studio Marketplace

Reviewer 2: there's a TODO link

do we know what URL this will eventually be based on what ID we'll release the extension under?

Reviewer 3: we should be able to: `https://marketplace.visualstudio.com/items?itemName=Shan.code-settings-sync` is an example of a page

Author: Fixed.

- The [source code](https://github.com/sorbet/sorbet/issues/new/choose) for the
  extension, located in the `vscode_extension/` folder of the Sorbet repo
- Full [installation and usage instructions](https://sorbet.org/docs/vscode) in
  the Sorbet docs

## Showcase

The best way to experience the Sorbet extension for VS Code is to try it out

> yourself. But to showcase a handful of its features, here are some screen

Reviewer 2: ```suggestion
yourself. To entice you to try it out for yourself, here are some screen
```

recordings.

When there are errors in a file, Sorbet will underline them in red. It also
shows a list of all errors in VS Code’s “Problems” window.

![Sorbet errors are shown with red squiggles in the editor. Hover to see the full message.](/img/lsp/errorsquiggle_blog.png)

Sorbet responds with autocompletion results while typing, including a list of
possible methods and the documentation for that item, including its signature.

![Autocompletion works for constants, methods, and variables](/img/lsp/autocomplete_blog.png)

Sorbet can find all references of a constant, method, or variable throughout an
entire codebase:

> <video autoplay muted loop width="100%" style="display:block;margin-left:auto;margin-right:auto;">
>     <source src="/img/lsp/references_blog.mp4" type="video/mp4">
> 
>     Sorry, your browser doesn't support embedded videos.
> 
> </video>
> 
> … and having found all those references, it can even rename them automatically:
> 
> <video autoplay muted loop width="100%" style="display:block;margin-left:auto;margin-right:auto;">
>     <source src="/img/lsp/rename_blog.mp4" type="video/mp4">
> 
>     Sorry, your browser doesn't support embedded videos.
> 
> </video>

Reviewer 2: ```suggestion
<p><video autoplay muted loop width="100%" style="display:block;margin-left:auto;margin-right:auto;">
    <source src="/img/lsp/references_blog.mp4" type="video/mp4">

    Sorry, your browser doesn't support embedded videos.

</video></p>

… and having found all those references, it can even rename them automatically:

<p><video autoplay muted loop width="100%" style="display:block;margin-left:auto;margin-right:auto;">
    <source src="/img/lsp/rename_blog.mp4" type="video/mp4">

    Sorry, your browser doesn't support embedded videos.

</video></p>
```

tested this live. it makes the spacing look nicer

There are also a handful of experimental features that we haven’t quite
stabilized yet. Feel free to try them out by enabling experimental LSP features

> by reading these docs. One of those features is the signature help tooltip, to

Reviewer 2: did we write these docs? we should either (write and) link to them or delete this reference.

Author: I did not document experimental features.

Author: I removed the blurb.

bring up the name and type of the current argument:

![Sorbet provides a signature helper tooltip while filling in a method’s arguments.](/img/lsp/experimental_blog.png)

## Acknowledgements

Finally, this release represents the work of many people over the last few
years, all of whom we’ve grateful for, including but not limited to:

- Ainsley Escorce-Jones
- Akshay Joshi
- Andrew O'Neil
- Anthony Pratti
- Ariel Davis
- Caleb Barde
- Christopher Brown
- Daniel Molina
- Dmitry Petrashko
- G. D. Ritter
- Ilya Zheleznikov
- Jacob Zimmerman
- Joey Pereira
- John Vilk
- Jonathan Fung
- Jérémie Laval
- Kevin Miller
- Lance Lafontaine
- Manjiri Tapaswi
- Mick Killianey
- Nathaniel Roman
- Nelson Elhage
- Patrick Vilhena
- Paul Tarjan
- Penelope Phippen
- Robb Shecter
- Soam Vasani
- Susan Tu
- Sushain Cherivirala
- Trevor Elliott
- Ufuk Kayserilioglu
- Vignesh Shankar

In particular, John Vilk has been instrumental in leading the development of the
Sorbet LSP server for the past several years. Thanks to him and all who have
contributed in the past!

As always, if you encounter issues or want to suggest features, file an issue
against

> [the Sorbet issue tracker](https://github.com/sorbet/sorbet/issues/new/choose)

Reviewer 2: Would it be useful for you to add an issue template for IDE bugs?

Author: Probably.

Author: I won't add it now. We should think about this and how it fits in w/ other issue templates.
